dali/pipeline/operator/op_schema_test.cc

```diff
@@ -51,28 +51,47 @@ TEST(OpSchemaTest, OutputFNTest) {
 
 DALI_SCHEMA(Dummy3)
   .NumInput(1).NumOutput(1)
-  .AddOptionalArg("foo", "foo", 1.5f);
+  .AddOptionalArg("foo", "foo", 1.5f)
+  .AddOptionalArg<int>("no_default", "argument without default", nullptr);
 
 TEST(OpSchemaTest, OptionalArgumentDefaultValue) {
   auto spec = OpSpec("Dummy3");
   auto &schema = SchemaRegistry::GetSchema("Dummy3");
 
   ASSERT_TRUE(schema.HasOptionalArgument("foo"));
-  ASSERT_EQ(schema.GetDefaultValueForOptionalArgument<float>("foo"), 1.5f);
+  ASSERT_EQ(schema.GetDefaultValueForArgument<float>("foo"), 1.5f);
+
+  ASSERT_TRUE(schema.HasOptionalArgument("no_default"));
+
+  ASSERT_TRUE(schema.HasArgumentDefaultValue("foo"));
+
+  ASSERT_FALSE(schema.HasArgumentDefaultValue("no_default"));
+
+  ASSERT_THROW(schema.HasArgumentDefaultValue("don't have this one"), std::runtime_error);
```

Comment:
What would return `schema.GetDefaultValueForArgument` for "no_default"? Maybe it would be add a test case too?

Reply:
It gets you an error:
```
Assert on "v != nullptr" failed: Optional argument "no_defualt" is not defined for schema "Dummy3"
```
I can add some tests.